### PR TITLE
feat: add Databricks ODBC engine spec

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Add an ODBC based Databricks DB engine spec. This allows Superset to connect to the new [SQL endpoints](https://docs.databricks.com/sql/admin/sql-endpoints.html).

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

I was able to connect Superset to a Databricks cluster via ODBC, as well as to a SQL endpoint.

Screenshot 2021-09-27 at 14-05-03  DEV  Superset

The steps to reproduce:

1. Follow the docs that I added. :)
2. Run a query.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/databricks.py [hunk 1]
```diff
@@ -15,11 +15,34 @@
...
+from typing import Optional
+
+from superset.db_engine_specs.base import BaseEngineSpec
 from superset.db_engine_specs.hive import HiveEngineSpec
 
 
 class DatabricksHiveEngineSpec(HiveEngineSpec):
     engine = "databricks"
-    engine_name = "Databricks"
+    engine_name = "Databricks (Hive)"
```
Comment (villebro) [inline superset/db_engine_specs/databricks.py:27]: nit: to avoid confusion, maybe this should be "Databricks (PyHive)"

**Review by villebro — APPROVED** [review]
Nice to see support added for the new SQL endpoints! 💯 One minor optional nit/comment, other than that LGTM

File: superset/db_engine_specs/databricks.py (hunk 1 above)
Comment (betodealmeida) [reply superset/db_engine_specs/databricks.py:27]: I talked with Eugenia and she suggested "Databricks Interactive Cluster" here, and "Databricks SQL Endpoint" for ODBC.

**Comment by bhavishya123** [comment]
@betodealmeida This feature is not available in latest superset pypi package - apache-superset 1.3.2
when it will be released ?

**Comment by betodealmeida** [comment]
> @betodealmeida This feature is not available in latest superset pypi package - apache-superset 1.3.2 when it will be released ?

@eschutho is currently working on getting 1.4.0 out, which should include this engine spec. Unfortunately we just found out a bug with 1.4.0rc3 and we're going to have to do an RC4, so it should take another couple of weeks.

**Pull request merged**